interfaces/builtin/serial_port.go

```diff
@@ -154,7 +156,7 @@ func (iface *SerialPortInterface) ConnectedPlugSnippet(plug *interfaces.Plug, sl
 		if iface.hasUsbAttrs(slot) {
 			// This apparmor rule must match serialDeviceNodePattern
 			// UDev tagging and device cgroups will restrict down to the specific device
-			return []byte("/dev/tty* rw,\n"), nil
+			return []byte("/dev/tty[A-Z]*[0-9] rw,\n"), nil
```

Comment:
I think you need to have a `+` at the end of the pattern, to match what is in the `serialDeviceNodePatten` above.